ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -262,6 +262,41 @@ let test_cbt =
       )
   in
 
+
+  let test_vdi_data_destroy () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+
+    (* change VDI properties so that data_destroy passes *)
+    let pass_data_destroy vdi =
+      let sr = Db.VDI.get_SR ~__context ~self:vdi in
+      Db.SR.set_is_tools_sr ~__context ~self:sr ~value:false;
+      Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:true;
+      Db.VDI.set_cbt_enabled ~__context ~self:vdi ~value:true in
+
+    (* change VDI properties one by one so data_destroy only fails for the indicated reason *)
+    List.iter (fun (vdi_fun, api_error) ->
+        run_assert_equal_with_vdi ~__context ~vdi_fun `data_destroy (Some api_error))
+
+      [
+        (fun vdi -> pass_data_destroy vdi;
+          Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:false
+        ) ,         (Api_errors.operation_not_allowed, ["VDI is not a snapshot"]) ;
+
+        (fun vdi -> pass_data_destroy vdi;
+          let sr = Db.VDI.get_SR ~__context ~self:vdi in
+          Db.SR.set_type ~__context ~self:sr ~value:"udev"
+        ) ,         (Api_errors.sr_operation_not_supported, ["VDI is a physical device"])  ;
```

Comment:
This should be removed, as we discussed. The new design says:
> The following check performed by VDI.destroy is also irrelevant,
because CBT is not supported on SRs of type "udev", therefore
CBT operations will return an SR_OPERATION_NOT_SUPPORTED
error instead:
· The VDI.destroy operation is not allowed on SRs of type
"udev", and returns the VDI_IS_A_PHYSICAL_DEVICE error
for such SRs.